osrf/vrx
# Ball shooter v2

Fixes #377.

This pull request updates the structure of the ball shooter to look realistic when a given pitch or yaw is specified.

![shooter_v2](https://user-images.githubusercontent.com/1440739/137555506-f773415f-5229-41ab-bb0c-cab31bc948ec.png)
>

How to test it:

* Launch the simulation
```
roslaunch vrx_gazebo vrx.launch
```

Observe that the ball shooter looks correct.

* Edit `wamv_gazebo/urdf/wamv_gazebo.urdf.xacro`, line 175 choosing your preferred pitch and yaw:

```

      <xacro:wamv_ball_shooter name="ball_shooter" position="0.54 0.30 1.296" pitch="-1.04"/>
```

Recompile the code and launch it again:

```
catkin_make
roslaunch vrx_gazebo vrx.launch
```

Observe that the ball shooter has been modified to your preferred orientation and still looks correct.

* Try firing a projectile:

```
rostopic pub /wamv/shooters/ball_shooter/fire std_msgs/Empty "{}" --once
```

Visually check that the ball follows the angle of the launcher.
